torch_model_base.py

```diff
@@ -26,6 +27,7 @@ def __init__(self,
             tol=1e-5,
             device=None,
             display_progress=True,
+            shuffle_train=True,
```

Comment:
It seems unusual to me to have the base class handle this detail. Is there a general motivation for it? If we do add it, it should be documented.